dotcom-rendering/src/web/components/ShowHideContainer.importable.tsx

```diff
@@ -0,0 +1,123 @@
+import { css } from '@emotion/react';
+import { from, space, neutral, remSpace } from '@guardian/source-foundations';
+import { ButtonLink } from '@guardian/source-react-components';
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+
+type Props = {
+	sectionId: string;
+	overrideContainerToggle: string | undefined;
+};
+
+type ContainerStates = {
+	value: { [id: string]: string };
+};
+
+const isServer = typeof window === 'undefined';
+
+const showHideButtonCss = (overrideContainerToggle: string | undefined) => {
+	return css`
+		color: ${neutral[46] || overrideContainerToggle};
+		line-height: ${remSpace[5]}rem;
+		font-size: 0.875rem;
+	`;
+};
+
+const showHideContainerCss = css`
+	margin-top: ${space[2]}px;
+	margin-right: 10px;
+	margin-bottom: ${space[2]}px;
+	position: relative;
+	align-items: bottom;
+
+	${from.wide} {
+		position: absolute;
+		top: 0;
+		right: 0;
+	}
+`;
+
+const getContainerStates = (): ContainerStates => {
+	const item = localStorage.getItem(`gu.prefs.container-states`) as
```

Comment:
We can use [libs/storage](https://github.com/guardian/libs/tree/main/src/storage) here 😎